encoding/gocode/decorators.go

```diff
@@ -0,0 +1,112 @@
+package gocode
+
+import (
+	"go/token"
+
+	"github.com/dave/dst"
+	"github.com/dave/dst/dstutil"
+)
+
+func isSingleTypeDecl(gd *dst.GenDecl) bool {
+	if gd.Tok == token.TYPE && len(gd.Specs) == 1 {
+		_, is := gd.Specs[0].(*dst.TypeSpec)
+		return is
+	}
+	return false
+}
+
+func isAdditionalPropertiesStruct(tspec *dst.TypeSpec) (dst.Expr, bool) {
+	strct, is := tspec.Type.(*dst.StructType)
+	if is && len(strct.Fields.List) == 1 && strct.Fields.List[0].Names[0].Name == "AdditionalProperties" {
+		return strct.Fields.List[0].Type, true
+	}
+	return nil, false
+}
+
+func DecoderCompactor() dstutil.ApplyFunc {
```

Comment:
Let's unexport this, i don't think there's value in exposing it

Reply:
That's true, thnx